packages/app/src/LogSidePanelElements.tsx

```diff
@@ -0,0 +1,333 @@
+      <div className="fs-7">{value}</div>
+    </div>
+  );
+};
+
+const StacktraceRowExpandButton = ({
+  onClick,
+  isOpen,
+}: {
+  onClick: VoidFunction;
+  isOpen: boolean;
+}) => {
+  return (
+    <TableCellButton
+      label={isOpen ? 'Hide context' : 'Show context'}
+      biIcon={isOpen ? 'chevron-up' : 'chevron-down'}
+      onClick={onClick}
+    />
+  );
+};
+
+export const StacktraceRow = ({ row }: { row: Row<StacktraceFrame> }) => {
+  const [lineContextOpen, setLineContextOpen] = React.useState(false);
+
+  const frame = row.original;
+  const hasContext = !!frame.context_line;
+
+  const handleToggleContext = React.useCallback(() => {
+    setLineContextOpen(!lineContextOpen);
+  }, [lineContextOpen]);
+
+  return (
+    <>
+      <div className="w-100 d-flex justify-content-between align-items-center">
+        <div>
+          {frame.filename}
+          <span className="text-slate-400">{' in '}</span>
+          {frame.function}
+          {frame.lineno || frame.colno ? (
+            <>
+              <span className="text-slate-400">{' at line '}</span>
+              <span className="text-slate-300">
+                {frame.lineno}:{frame.colno}
+              </span>
+            </>
+          ) : null}
+        </div>
+        {hasContext && (
+          <StacktraceRowExpandButton
+            onClick={handleToggleContext}
+            isOpen={lineContextOpen}
+          />
+        )}
+      </div>
+
+      {lineContextOpen && (
+        <pre className={styles.lineContext}>
+          {frame.pre_context?.map((line, i) => (
+            <div key={line}>
+              <span className={styles.lineContextLineNo}>
+                {(frame.lineno ?? 0) - (frame.pre_context?.length ?? 0) + i}
+              </span>
+              {line}
+            </div>
+          ))}
+          {frame.context_line && (
+            <div className={styles.lineContextCurrentLine}>
+              <span className={styles.lineContextLineNo}>{frame.lineno}</span>
+              {frame.context_line}
+            </div>
+          )}
+          {frame.post_context?.map((line, i) => (
+            <div key={line}>
+              <span className={styles.lineContextLineNo}>
+                {frame.lineno + i + 1}
+              </span>
+              {line}
+            </div>
+          ))}
+        </pre>
+      )}
+    </>
+  );
+};
+
+export const stacktraceColumns: ColumnDef<StacktraceFrame>[] = [
+  {
+    accessorKey: 'filename',
+    cell: StacktraceRow,
+  },
+];
+
+export const breadcrumbColumns: ColumnDef<StacktraceBreadcrumb>[] = [
+  {
+    accessorKey: 'category',
+    header: 'Category',
+    cell: ({ row }) => (
+      <span className="text-slate-300">{row.original.category}</span>
+    ),
+  },
+  {
+    accessorKey: 'message',
+    header: 'Message',
+    size: UNDEFINED_WIDTH,
+    cell: ({ row }) =>
+      row.original.message || <span className="text-slate-500">Empty</span>,
```

Comment:
I think it'd be nice here if can also make use of the fetch/xhr categories, as they currently show up as empty (maybe a follow up PR?) Here's a few examples if helpful of the data structure I see:

```
{
  "category": "fetch",
  "data": {
    "method": "GET",
    "status_code": 200,
    "url": "http://localhost:8080/__nextjs_original-stack-frame"
  },
  "timestamp": 1700379487.918,
  "type": "http"
}
```

```
{
  "category": "xhr",
  "data": {
    "method": "POST",
    "status_code": 200,
    "url": "http://localhost:4318/v1/traces"
  },
  "timestamp": 1700379491.015,
  "type": "http"
}
```

Reply:
I'll do it in a separate pr